actor/src/lib.rs

```diff
@@ -160,12 +159,15 @@ impl<A: Actor> Addr<A> {
     {
         let envelope = SyncEnvelopeProxy::pack(message, None);
         let sender = self.sender.clone();
-        tokio::task::spawn(async move {
-            // TODO: propagate the error.
-            if let Err(error) = sender.send(envelope).await {
-                iroha_logger::error!(%error, "Error sending actor message");
+        // TODO: BUG: remove deadlock from iroha (probably issue inside of `iroha_p2p` crate) and remove this task::spawn
+        tokio::spawn(
+            async move {
+                if let Err(error) = sender.send(envelope).await {
+                    iroha_logger::error!(%error, "Error sending actor message");
+                }
             }
-        });
+            .in_current_span(),
+        );
```

Comment:
so nothing changed in this method? except for the added tracing?

Reply:
Exactly